test/extensions/common/wasm/test_data/test_rust.rs

```diff
@@ -1,5 +1,6 @@
 // Build using:
 // $ rustc -C lto -C opt-level=3 -C panic=abort -C link-arg=-S -C link-arg=-zstack-size=32768 --crate-type cdylib --target wasm32-unknown-unknown test_rust.rs
+// $ ../../../../../bazel-bin/test/tools/wee8_compile/wee8_compile_tool test_rust.wasm test_rust.wasm
```

Comment:
It does not thrill me that we are checking in pre-compiled artifacts to run tests. Is there any reason this cannot be done as part of the build process so it's hermetic and it's easier to make changes to the test code? I suspect as we add more ABI functionality we will keep having more test code. I don't want their to be a separate dev step in order to write the test code in all the various languages that we will have.

Reply:
Neither am I, but we already have a precedent for that (TLS certificates).

In any case, both me & @jplevyak are working on making this hermetic, but it's a slow process with many moving parts, and I'd rather not block all the Wasm contributions on the hermetic generation of test data.

Reply:
Yes, agreed it's fine to do as a follow up. Let's please start opening upstream issues which we will tag with the area/wasm label for follow up. Let's also mention those issues in TODOs in the code. This PR needs 3 issues opened and tagged in TODOs:
1) Add WASM module loading fuzzing
2) Upstream wee8_compile tool into v8
3) Make all WASM tests fully hemertic

Thank you.

/wait

Reply:
Done (#9731, #9732, #9733).

Reply:
Can you please add relevant TODOs to the code referencing these issues and we can ship? Thank you.

/wait

Reply:
I added TODOs for #9731 and #9733 to the code.

I didn't add TODO to upstream the tool to V8 repo (#9732), since I tried to upstream it already, and the V8's Wasm team believes that this tool is too specific to our use case, and don't think it should live in the V8 repo.

I can move it to a separate repo if you don't want it here, though. Let me know.

Reply:
Let's move this discussion about the repo to #9732.